engine/samples/imgui/page.md

```diff
@@ -0,0 +1,27 @@
+# ImGui {#examples-engine-imgui}
+
+@brief Using the @ref imgui-plugin plugin.
+
+The ImGui plugin allows you to integrate the [Dear ImGui](https://github.com/ocornut/imgui) library with CUBOS. engine, making it quick, straightforward, and effortless to create user interfaces and interactive debugging tools within your applications.
+
+![](imgui/img1.png)
+
+First, let's start by including the header files we need.
+
+@snippet imgui/main.cpp Including plugin headers
+
+Then, make sure to add the plugin.
+
+@snippet imgui/main.cpp Adding the plugin
+
+Once the ImGui plugin is added, you can create systems to display ImGui windows and widgets. Here's an example of how to create an ImGui window.
+
+@snippet imgui/main.cpp ImGui window example
+
+Ensure that you add your system with the `cubos.imgui` tag; otherwise, the ImGui elements from that system won't be be visible.
+
+@snippet imgui/main.cpp Adding the system
+
+Pretty simple right? You're now equipped to craft and utilize ImGui's functions to design your cool user interface.
```

Comment:
It'd be nice if we had a link here to where people can check how to use imgui stuff. Someone coming to this sample is probably looking for information that is more in depth than just the process of spawning a window.

Reply:
Well for that we could just add ImGui's documentation link or something.

Reply:
Yeah, that was what I meant, sorry if I wasn't clear, it wasn't telling you to document all of imgui yourself